### DISCUSSION TITLE
Role-Based Access Control (RBAC) for Self-hosted | Beginner Questions

### DISCUSSION THREAD

**Discussion by rexhibition** [supabase/supabase#22272] [Questions]
I am a beginner and I have been sleepless to get this whole things to work. But it's just not possible. I am so furstrated so if anyone can help, I would be deeply grateful for it.

What I did:
- Clone this [Slack-clone](https://github.com/supabase/supabase/tree/master/examples/slack-clone/nextjs-slack-clone).
- Paste the "[Create Auth Hook to Apply User Role](https://supabase.com/docs/guides/auth/custom-claims-and-role-based-access-control-rbac#create-auth-hook-to-apply-user-role)" codes to the SQL Editor on Supabase.

- Enable the hook... This part I saw the config,toml file already have the code so I didn't do anything.

![image](https://github.com/supabase/supabase/assets/64733112/3cee8783-b6f4-4cff-b8f2-f34e182b151a)

- Finally, add the codes from the "Accessing Custom Claim in RLS policies" into the SQL Editor on Supabase.

The last step seems to already be done so I didn't do anything. I run the app and the register, sign in, write text, etc. work except that the app doesn't recognize the role so it doesn't allow admin or moderator account to delete anything (not even the icon is out). The token in the log doesn't have any custom claim in it.

I am very desparate now. Please help :(

**Comment by barrownicholas** [comment]
Did you ever get this to work? Running in to this now

**Reply by rexhibition** [reply]
Unfortunately not. I ended up doing this instead: https://github.com/supabase-community/supabase-custom-claims
